src/components/tutorials-landing-view/collection-content-card-link/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { type CollectionContentCardLinkProps } from '../types'
+import TutorialsLandingContentCardLink from '../content-card-link'
+import boundaryGraphic from './img/boundary.svg'
+import consulGraphic from './img/consul.svg'
+import nomadGraphic from './img/nomad.svg'
+import packerGraphic from './img/packer.svg'
+import terraformGraphic from './img/terraform.svg'
+import vagrantGraphic from './img/vagrant.svg'
+import vaultGraphic from './img/vault.svg'
+import waypointGraphic from './img/waypoint.svg'
+
+const PRODUCT_SLUGS_TO_HEADER_IMAGES = {
+	boundary: boundaryGraphic,
+	consul: consulGraphic,
+	nomad: nomadGraphic,
+	packer: packerGraphic,
+	terraform: terraformGraphic,
+	vagrant: vagrantGraphic,
+	vault: vaultGraphic,
+	waypoint: waypointGraphic,
+}
+
+const CollectionContentCardLink = ({
+	badges,
+	description,
+	href,
+	productSlug,
+	title,
+	tutorialCount,
+}: CollectionContentCardLinkProps) => {
```

Comment:
I think I recommend here just passing this a collection object directly, as this would make the ergonomics of this component super simple, and reduce the burden on the consumer.  For example, right now the consumer of this component needs to know the logic of what badges go into this component, whereas we can actually deterministically coerce a collection object into those badges.

While View Model components decouple the presentation from the domain, Domain Model components can decouple the domain from the presentation.

Reply:
In this case I chose not to pass `collection` so that we can do all of our data shaping server-side rather than having the view or its subcomponents do it client-side. If a collection object had all the properties we need for rendering these cards directly accessible, without needing any `map`s or boolean comparisons, then I would choose to pass that object. But because we need to `map` over all the tutorials for a collection to get the unique set of badges, I think it's better for us to do that shaping in the same place we're fetching data and passing it into the view in the shape it needs.

Reply:
>  I think it's better for us to do that shaping in the same place we're fetching data

I may possibly have objections, but I'm looking to understand your perspective further / understand the motivation here 

> If a collection object had all the properties we need for rendering these cards directly accessible, without needing any maps or boolean comparisons, then I would choose to pass that object.

Is this related to a performance concern?  What's the downside if you had to do the mapping / doing the comparisons on the client side?

Reply:
Performance & separation of concerns -- it's not necessary for the component to do this mapping every time it renders, and it if it doesn't _have_ to be concerned about the source data shape it doesn't feel super necessary here.

Reply:
Got it, so performance is something that could influence my thoughts here, but only _after_ we have run into a performance issue.  Something to definitely keep in mind in the event this comes up.

Regarding separation of concerns:

I guess we would have to first define what the concern of this layer is. In my perspective, a domain model component's _sole_ responsibility is to _map the domain to the presentation_.  I'm curious if you have a different definition of what the concern of this layer is responsible for -- that would further help me understand your perspective here!

With the approach that we are currently taking, it makes me not understand the responsibility of this layer. If we plan on mapping the `collection` to this interface at the data fetching layer, we could just also just map it to the View Model's shape.  With that approach, at least all of the business mapping logic would be in once place.

My biggest concern is actually though the data fetching layer _also_ be responsible for mapping the content to its presentational shape. In the future, if we wanted to adjust the card to also display something else, or we wanted to update the business logic to render a new badge, we would need to update the data fetching layer.

Reply:
@BrandonRomano have updated `CollectionContentCardLink` to just accept 2 props now: `productSlug` and `collection`. We've not consistently broken things up into this many layers and also try to minimize the amount of data passed from the server to the client (hence going ahead and shaping it if we're manipulating it anyways), but we don't have enough time to dig in right now anyways so I've made the changes. Let me know what you think!